### PR TITLE
fix(manager/gitlabci): regex for services

### LINKED ISSUE #14765: Gitlab: Image with quotes failed to parse
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.6.10

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

14.8

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

service images with quotes are not parsed correctly any more:

Minimal example:
```
services:
  - "mariadb:10.4.11"
```
Gets parsed as `mariadb:10.4.11"`.

### Relevant debug logs

<details><summary>Logs</summary>

```
Object {
              "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
              "currentDigest": undefined,
              "currentValue": "10.4.11\"",
              "datasource": "docker",
              "depName": "mariadb",
              "replaceString": "mariadb:10.4.11\"",
            }
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

### PR DESCRIPTION
**Pull request by hasanwhitesource**
## Changes

regex matching image pattern

## Context

Closes #14765 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by hasanwhitesource** [comment]
@viceice I've added tests to cover the common cases that I could think of. Do you have any more cases to suggest ?

File: lib/modules/manager/gitlabci/extract.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ const imageRe = regEx(
 );
 const nameRe = regEx(`^\\s*name:\\s+['"]?(?<depName>[^\\s'"]+)['"]?\\s*$`);
 const serviceRe = regEx(
-  `^\\s*-?\\s*(?:name:\\s+)?['"]?(?<depName>[^\\s'"]+[^:]$)['"]?\\s*$`
+  `^\\s*-?\\s*(?:name:\\s+)?['"]?(?<depName>[\\{\\}\\$\\w.\\d\\/\\@]+:[\\{\\}\\$\\w.\\d\\:\\@-]+)['"]?\\s*$`
```
Comment (viceice) [inline lib/modules/manager/gitlabci/extract.ts:19]: Why do we need such a complex regex?

I really would prefer to use yaml based extraction, which would be more safe.

Comment (dsch) [reply lib/modules/manager/gitlabci/extract.ts:19]: I think even this regex could fail for multiple services:
```yaml
job1:
  services:
    - name: mooseagency/postgresql:12.3-1@sha256:a5a65569456f221ee1f8a0b3b4e2d440eb5830772d9440c9b30b1dbfd454c778
      command:
        - something:thatIsNotAnImage
        - -something2
        - -something3
        - -something4
    - "mariadb:10.4.11"
    - postgres:11.7
    - redis:latest
    - name: "registry.example.com/myimage:latest"
    - myimage@sha256:0ecb2ad60
    - tomcat:7-jre8
```

NB: Here you can find the schema for the gitlab-ci.yml: https://gitlab.com/gitlab-org/gitlab/-/raw/master/app/assets/javascripts/editor/schema/ci.json

Comment (hasanwhitesource) [reply lib/modules/manager/gitlabci/extract.ts:19]: @viceice @rarkins I will do more research on this and change the method that is based on regex parsing

**Comment by rarkins** [comment]
Let's instead solve this by refactoring the manager to use YAML parsing. New issue: https://github.com/renovatebot/renovate/issues/14843

**Pull request closed without merging**